src/main/java/net/minestom/server/utils/PacketUtils.java

```diff
@@ -101,12 +102,24 @@ public static void sendPacket(@NotNull Audience audience, @NotNull ServerPacket
      */
     public static void sendGroupedPacket(@NotNull Collection<Player> players, @NotNull ServerPacket packet,
                                          @NotNull Predicate<Player> predicate) {
-        final SendablePacket sendablePacket = GROUPED_PACKET ? new CachedPacket(packet) : packet;
+        final var sendablePacket = shouldUseCachePacket(packet) ? new CachedPacket(packet) : packet;
+
         players.forEach(player -> {
             if (predicate.test(player)) player.sendPacket(sendablePacket);
         });
     }
 
+    /**
+     * Checks if the {@link ServerPacket} is suitable to be wrapped into a {@link CachedPacket}.
+     * Note: {@link ComponentHoldingServerPacket}s are not translated inside a {@link CachedPacket}.
+     *
+     * @see CachedPacket#body()
+     * @see PlayerSocketConnection#writePacketSync(SendablePacket, boolean)
+     */
+    private static boolean shouldUseCachePacket(final @NotNull ServerPacket packet) {
+        return (!MinestomAdventure.AUTOMATIC_COMPONENT_TRANSLATION || !(packet instanceof ComponentHoldingServerPacket)) && GROUPED_PACKET;
```

Comment:
May be worth checking if the `ComponentHoldingServerPacket` has any component to actually translate. For example in packets containing items where you may not have to do anything

Reply:
I'm not sure, if the way I implemented it is the best.